exec/java-exec/src/main/java/org/apache/drill/exec/expr/fn/impl/CryptoFunctions.java

```diff
@@ -0,0 +1,389 @@
+
+    @Param
+    VarCharHolder rawInput;
+
+    @Output
+    VarCharHolder out;
+
+    @Inject
+    DrillBuf buffer;
+
+    @Override
+    public void setup() {
+
+    }
+
+    @Override
+    public void eval() {
+
+      String input = org.apache.drill.exec.expr.fn.impl.StringFunctionHelpers.toStringFromUTF8(rawInput.start, rawInput.end, rawInput.buffer);
+
+      String sha384 = org.apache.commons.codec.digest.DigestUtils.sha384Hex(input);
+
+      out.buffer = buffer;
+      out.start = 0;
+      out.end = sha384.getBytes().length;
+      buffer.setBytes(0, sha384.getBytes());
+    }
+
+  }
+
+  /**
+   * This function returns the SHA512 digest of a given input string.
+   *  Usage is shown below:
+   *  select sha512( 'testing' ) from (VALUES(1));
+   */
+
+
+  @FunctionTemplate(name = "sha512", scope = FunctionTemplate.FunctionScope.SIMPLE, nulls = FunctionTemplate.NullHandling.NULL_IF_NULL)
+  public static class SHA512Function implements DrillSimpleFunc {
+
+    @Param
+    VarCharHolder rawInput;
+
+    @Output
+    VarCharHolder out;
+
+    @Inject
+    DrillBuf buffer;
+
+    @Override
+    public void setup() {
+
+    }
+
+    @Override
+    public void eval() {
+
+      String input = org.apache.drill.exec.expr.fn.impl.StringFunctionHelpers.toStringFromUTF8(rawInput.start, rawInput.end, rawInput.buffer);
+
+      String sha512 = org.apache.commons.codec.digest.DigestUtils.sha512Hex(input);
+
+      out.buffer = buffer;
+      out.start = 0;
+      out.end = sha512.getBytes().length;
+      buffer.setBytes(0, sha512.getBytes());
+    }
+
+  }
+
+  /**
+   * aes_encrypt()/ aes_decrypt(): implement encryption and decryption of data using the official AES (Advanced Encryption Standard) algorithm,
+   * previously known as “Rijndael.” AES_ENCRYPT() encrypts the string str using the key string key_str and returns a
+   * binary string containing the encrypted output.
+   * Usage:  SELECT aes_encrypt( 'encrypted_text', 'my_secret_key' ) AS aes FROM (VALUES(1));
+   */
+
+
+  @FunctionTemplate(name = "aes_encrypt", scope = FunctionTemplate.FunctionScope.SIMPLE, nulls = FunctionTemplate.NullHandling.NULL_IF_NULL)
+  public static class AESEncryptFunction implements DrillSimpleFunc {
+
+    @Param
+    VarCharHolder rawInput;
+
+    @Param
+    VarCharHolder rawKey;
+
+    @Output
+    VarCharHolder out;
+
+    @Inject
+    DrillBuf buffer;
+
+    @Workspace
+    Cipher cipher;
+
+    @Override
+    public void setup() {
+      String key = org.apache.drill.exec.expr.fn.impl.StringFunctionHelpers.toStringFromUTF8(rawKey.start, rawKey.end, rawKey.buffer);
+      java.security.MessageDigest sha = null;
+      try {
+        byte[] keyByteArray = key.getBytes("UTF-8");
+        sha = java.security.MessageDigest.getInstance("SHA-1");
+        keyByteArray = sha.digest(keyByteArray);
+        keyByteArray = java.util.Arrays.copyOf(keyByteArray, 16);
+        SecretKeySpec secretKey = new SecretKeySpec(keyByteArray, "AES");
```

Comment:
@arina-ielchiieva I'm not sure.  Is there a style or functional reason to use the direct imports?